modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,237 @@
+# frozen_string_literal: true
+
+require 'sentry_logging'
+
+module Mobile
+  module V0
+    module Adapters
+      class AppointmentRequests
+        # accepts an array of appointment requests
+        # returns a list of appointments, filtering out any that are not SUBMITTED or CANCELLED
+        def parse(requests)
+          va_appointments = []
+          cc_appointments = []
+
+          requests.each do |request|
+            status = status(request)
+            next unless status.in?(%w[CANCELLED SUBMITTED])
+
+            if request.cc_appointment_request
+              cc_appointments << build_appointment_model(request, CC)
+            else
+              va_appointments << build_appointment_model(request, VA)
+            end
+          end
+
+          [va_appointments, cc_appointments]
+        end
+
+        private
+
+        # rubocop:disable Metrics/MethodLength
+        def build_appointment_model(request, klass)
+          start_date = localized_start_date(request)
+
+          Mobile::V0::Appointment.new(
+            id: request[:appointment_request_id],
+            appointment_type: appointment_type(request, klass),
+            cancel_id: nil,
+            comment: nil,
+            facility_id: klass.facility_id(request),
+            sta6aid: nil,
+            healthcare_provider: klass.provider_name(request),
+            healthcare_service: klass.practice_name(request),
+            location: klass.location(request),
+            minutes_duration: nil,
+            phone_only: phone_only?(request),
+            start_date_local: start_date,
+            start_date_utc: start_date.utc,
+            status: status(request),
+            status_detail: status_detail(request),
+            time_zone: time_zone(request),
+            vetext_id: nil,
+            reason: request[:purpose_of_visit],
+            is_covid_vaccine: nil,
+            is_pending: true,
+            proposed_times: proposed_times(request),
+            type_of_care: request[:appointment_type],
+            patient_phone_number: request[:phone_number],
+            patient_email: request[:email],
+            best_time_to_call: request[:best_timeto_call],
+            friendly_location_name: request[:friendly_location_name]
+          )
+        end
+        # rubcop:enable Metrics/MethodLength
+
+        def appointment_type(request, klass)
+          # this is temporary because test data does not include video type
+          unless request[:visit_type].in?(['Office Visit', 'Express Care', 'Phone Call'])
+            log_message_to_sentry('Unknown appointment request type', :error, { visit_type: request[:visit_type] })
+          end
+          klass::APPOINTMENT_TYPE
+        end
+
+        def phone_only?(request)
+          request[:visit_type] == 'Phone Call'
+        end
+
+        def proposed_times(request)
+          {
+            option_date1: request[:option_date1] == 'No Date Selected' ? nil : request[:option_date1],
```

Comment:
I know the VAMF service returns it like this but maybe we could have proposed times be an array of hashes? Is each time a specific time or like 'morning/afternoon', or 'from 2-4pm'?

Reply:
I'm happy to change the format. I'll let Ken know that's coming. All the data I've seen just says "AM" or "PM".

Reply:
👍 I think that future proofs it, and the FE can also just iterate over the list. I've also seen some option_17 option_18 etc in other government APIs 😬

Reply:
If it's anything other than AM/PM it'll cause issues because we're using proposed times for sorting purposes. But we'll deal with those errors if they arise.